web/yo/app/scripts/directives/addmodifytherapy.js

```diff
@@ -0,0 +1,171 @@
+'use strict';
+angular.module('oncokbApp')
+    .directive('addModifyTherapy', function ($rootScope, DatabaseConnector, _, $q, FirebaseModel, mainUtils, $window) {
+        return {
+            templateUrl: 'views/addModifyTherapy.html',
+            restrict: 'E',
+            controller: function ($scope) {
+                var therapyUuid = [];
+                if ($scope.modifyMode === true) {
+                    clearData();
+                    initTherapy();
+                } else {
+                    clearData();
+                }
+
+                function clearData() {
+                    $scope.therapy = [[]];
+                    therapyUuid = [[]];
+                    $scope.addTherapyError = false;
+                    $scope.noData = true;
+                    $scope.therapyResult = "";
+                }
+
+                function refreshResult() {
+                    $scope.therapyResult = _.map($scope.therapy, function (element) {
+                        return (_.map(element, function (name) {
+                            return name.drugName;
+                        }).join(" + "));
+                    }).join(", ");
+                }
+
+                function initTherapy() {
+                    var newTherapy = [];
+                    newTherapy = mainUtils.therapyStrToArr($scope.treatmentRef.name);
+                    firebase.database().ref('Drugs').once('value').then(function (snapshot) {
```

Comment:
Why do you use `once` instead of `on`? If you use `on`, you can keep listen on the Drugs collection and refresh $scope.drugList.

Reply:
I see. Indeed, once is better than on in here.